gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/db/FabricDatabase.java

```diff
@@ -857,8 +858,18 @@ private static final void dropTables(EmbedConnection embedConn,
           // make sure that corresponding row buffer also does not contain data
           final Region<?, ?> rowBuffer =
               Misc.getRegionForTable(tableName, false);
-          boolean tableContainsData = (columnBuffer.size() != 0) || (rowBuffer != null && rowBuffer.size() != 0);
-          if (!tableContainsData || (tableContainsData && removeTablesWithData)) {
+          String reservoirRegionName = Misc.getReservoirRegionNameForSampleTable(schema, tableName);
+          Region<Object, Object> reservoirRegion = Misc.getRegionForTable(reservoirRegionName,
+              false);
+          boolean tableContainsData = (columnBuffer.size() != 0) || (rowBuffer != null && rowBuffer.size() != 0)
+              || (reservoirRegion != null && reservoirRegion.size() != 0);
```

Comment:
I think size check should be removed. The reservoir could have been completely rolled into column store and may not have anything but we still want to remove the region. You can check for !isDestroyed()

@dshirish Not sure why size checks are there for columnBuffer/rowBuffer regions. Either could be zero sized but they need to be cleaned from DataDictionary regardless. Could test for !isDestroyed()

Reply:
@sumwale This was just to be on safe side to avoid dropping tables that contain data etc. The repair_catalog() proc by default does not drop a table if columnBuffer / rowBuffer contains data.  The procedure accepts a boolean parameter,  `removeTablesWithData` that allows removing the tables if they contain data. If this is false, just logs a warning.  Could be removed if this is not of much use.

Reply:
@dshirish This method will only be called if "removeTablesWithData" is true, right? In that case its quite possible that rowBuffer table size is 0 because all data has folded into column store, so then this check will skip deleting this region. Won't that cause problems when user tries to recreate the table failing with region already exists?

Reply:
@sumwale The method (FabricDatabase#dropTables) is always called even if "removeTablesWithData" is false.  The "tableContainsData" flag is set after checking sizes of both columnBuffer and rowBuffer and will skip dropping it if either of the size is non zero and will proceed to drop the table only if user has specified so using "removeTablesWithData". Hope I am following the question.

Reply:
@sumwale If column store is skipped "columnBatchTableExists" flag is set to true to skip dropping rowBuffer as well.

Reply:
ok